Public/Sdk/Public/Prelude/Prelude.AmbientHacks.dsc

```diff
@@ -95,12 +95,14 @@ namespace _PreludeAmbientHack_Transformer {
     export declare function getIpcServerMoniker(): IpcMoniker;
     export declare function getDominoIpcServerMoniker(): IpcMoniker;
     export declare function copyFile(sourceFile: File, destinationFile: Path, tags?: string[], description?: string, keepOutputsWritable?: boolean): DerivedFile;
+    
     export declare function writeFile(destinationFile: Path, content: any, tags?: string[], separator?: string, description?: string): DerivedFile;
-    export declare function writeData(destinationFile: Path, content: any, tags?: string[], description?: string): DerivedFile;
-    export declare function writeAllLines(destinationFile: Path, contents: any[], tags?: string[], description?: string): DerivedFile;
-    export declare function writeAllText(destinationFile: Path, content: string, tags?: string[], description?: string): DerivedFile;
-    export declare function sealDirectory(root: Directory, files: File[], tags?: string[], description?: string, scrub?: boolean): FullStaticContentDirectory;
-    export declare function sealSourceDirectory(root: Directory, option?: number, tags?: string[], description?: string, patterns?: string[]): SourceDirectory;
-    export declare function sealPartialDirectory(root: Directory, files: File[], tags?: string[], description?: string): PartialStaticContentDirectory;
-    export declare function composeSharedOpaqueDirectories(root: Directory, directories: SharedOpaqueDirectory[]): SharedOpaqueDirectory;
+    export declare function writeData(destinationPathOrArgs: (Path | Object), content: any, tags?: string[], description?: string): DerivedFile;

```

Comment:
why is the type here `Object` and not some new interface with concrete properties for args?

Reply:
Since our current implementation requires ambients to be implemented in Prelude and everythign in prelude needs to be implemented in C#, this is just a hacky bridge between some modules and the prelude. No users should be calling this api directy as evident by the namespace. We have work to do to update how ambients are implemented and make that version aware as well. The type checking happens at the code calling this.

---
In reply to: [272252535](https://github.com/Microsoft/BuildXL/pull/82#discussion_r272252535) [](ancestors = 272252535)